File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> seen = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int key = target-nums[i];
            if(seen.containsKey(key)){
                return new int[]{seen.get(key),i};
            }
            seen.put(nums[i],i);
        }

        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen: HashMap<i32,i32> = HashMap::new();

        for(i,&num) in nums.iter().enumerate() {
            let comp = target - num;
            if let Some(&id) = seen.get(&comp){
                return vec![id,i as i32];
            }
            seen.insert(num,i as i32);
        }
        vec![-1,-1]
    }
}